heron/packing/src/java/com/twitter/heron/packing/Container.java

```diff
@@ -59,14 +65,53 @@ private boolean hasSpace(long ram, double cpuCores, long disk) {
    *
    * @return true if the instance can be added to the container, false otherwise
    */
-  public boolean add(Resource resource) {
+  public boolean add(Resource resource, InstanceId instanceId) {
     if (this.hasSpace(resource.getRam(), resource.getCpu(), resource.getDisk())) {
       usedRam += resource.getRam();
       usedCpuCores += resource.getCpu();
       usedDisk += resource.getDisk();
+      instanceIds.add(instanceId);
       return true;
     } else {
       return false;
     }
   }
+
+  /**
+   * Remove an instance of a particular component from a container and update its
+   * corresponding resources.
+   *
+   * @return instanceId if the instance is removed the container.
+   * Return null if the container is empty or
+   * does not contain an instance of the given component.
+   */
+  public InstanceId remove(Resource resource, String component) {
+    if (!instanceIds.isEmpty()) {
+      InstanceId instanceId = instanceOfComponent(component);
+      if (instanceId != null) {
+        usedRam -= resource.getRam();
+        usedCpuCores -= resource.getCpu();
+        usedDisk -= resource.getDisk();
+        instanceIds.remove(instanceId);
+        return instanceId;
+      } else {
+        return null;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Find whether an instance of a particular component is assigned to the container
+   *
+   * @return the instanceId if an instance is found, null otherwise
+   */
+  public InstanceId instanceOfComponent(String component) {
```

Comment:
Yes, but this method returns only one of them. I don't care about returning all of them. ANy one would work. How about public InstanceId getAnInstanceOfComponent(String component)?